processor/spanprocessor/config.go

```diff
@@ -16,12 +16,36 @@ package spanprocessor
 
 import (
 	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/processor/common"
 )
 
 // Config is the configuration for the span processor.
+// Prior to any actions being applied, each span is compared against
+// the include properties and then the exclude properties if they are specified.
+// This determines if a span is to be processed or not.
 type Config struct {
 	configmodels.ProcessorSettings `mapstructure:",squash"`
 
+	// Include specifies the set of span properties that must be present in order
+	// for this processor to apply to it.
+	// Note: If `exclude` is specified, the span is compared against those
+	// properties after the `include` properties.
+	// This is an optional field. If neither `include` and `exclude` are set, all spans
+	// are processed. If `include` is set and `exclude` isn't set, then all
+	// spans matching the properties in this structure are processed.
+	Include *common.MatchProperties `mapstructure:"include"`
```

Comment:
The two alternatives below look a bit better to me.
```golang
Include *spanmatcher.Config `mastructure:"include"`
```
```golang
Include *span.MatcherConfig `mastructure:"include"`
```

Reply:
So I moved the include and exclude to be under MatchConfig.